sdk/storage/src/blob/blob/requests/put_block_blob_builder.rs

```diff
@@ -47,7 +47,7 @@ impl<'a> PutBlockBlobBuilder<'a> {
         content_language: ContentLanguage<'a> => Some(content_language),
         content_disposition: ContentDisposition<'a> => Some(content_disposition),
         metadata: &'a Metadata => Some(metadata),
-        access_tier: AccessTier => access_tier,
+        access_tier: Option<AccessTier> => access_tier,
```

Comment:
This should be:

```rust
access_tier: AccessTier => Some(access_tier)
```

Reply:
Hi @MindFlavor, I had tried that originally (see commit 29a54fb388) but it didn't work:

```
error[E0277]: the trait bound `AccessTier: From<std::option::Option<_>>` is not satisfied
   --> src/kv/azure.rs:228:14
    |
228 |             .access_tier(None)
    |              ^^^^^^^^^^^ the trait `From<std::option::Option<_>>` is not implemented for `AccessTier`
    |
    = note: required because of the requirements on the impl of `Into<AccessTier>` for `std::option::Option<_>`

error: aborting due to previous error
```

Reply:
mmm it seems we are talking about two different issues. 
In order for the code to compile the macro `setters!` expects the abovementioned change (the one you originally put in commit [https://github.com/Azure/azure-sdk-for-rust/commit/29a54fb388e1584d6fd08e712696e9c52a9d25a7](https://github.com/Azure/azure-sdk-for-rust/commit/29a54fb388e1584d6fd08e712696e9c52a9d25a7)). 

As for your code, there is no need to specify `None` as access tier since it's the default. In fact, you cannot specify `None` as you found out.

The (current) logic of builder is: if an option is not specified the default value is is passed downstream. If the there is default value, the header is not passed at all. If you specify a value, the header will be passed downstream.

The bug in #165 here was to imply that `AccessTier::Hot` was a valid default, which is not if we use the same client for every storage type.

Reply:
Ah, that makes sense now. I've applied and pushed the change.